gurklang/stdlib_modules/strings.py

```diff
@@ -1,13 +1,74 @@
-from typing import Callable, TypeVar, Tuple
-from ..vm_utils import render_value_as_source
+from typing import TypeVar, Tuple
+
 from ..builtin_utils import BuiltinModule, Fail, make_function, raw_function, make_simple
-from ..types import CallByName, CallByValue, NativeFunction, Put, State, Str, Value, Stack, Scope, Int
-import random
+from ..types import CallByValue, Put, State, Str, Value, Stack, Int, Atom
+from ..vm_utils import render_value_as_source
 
 module = BuiltinModule("strings")
 T, V, S = Tuple, Value, Stack
 Z = TypeVar("Z", bound=Stack)
 
+_UNARY_STRING_TO_STRING = [
+    ('casefold', 'fold-case'),
+    ('lower', '->lower'),
+    ('upper', '->upper'),
+    ('swapcase', 'swap-case'),
+    ('title', '->title'),
+]
+_UNARY_STRING_TO_BOOL = [
+    ('is' + w, w + '?')
+    for w in 'alnum alpha ascii decimal digit identifier lower numeric printable space title upper'.split()
+]
+
+
+def _register_delegated_methods():
+    for function, name in _UNARY_STRING_TO_STRING:
+        method = getattr(str, function)
+
+        def fun(stack: T[V, S], fail: Fail, method=method):
```

Comment:
I would add a comment referencing the closure gotcha to these functions, because it's not immediately obvious why the default parameter is there